### DISCUSSION TITLE
git squash 사용에 대해

### DISCUSSION THREAD

**Discussion by salmontaker** [wanted-pre-onboarding-team-12th-7/pre-onboarding-12th-1-7#22] [Ideas]
이번주는 저희가 회의에서 정한대로 merge 방식으로 커밋을 진행하였는데요,

저희가 의도했던대로 작업단위가 나뉘어져서 어떤 작업을 했는지 히스토리를 통해 파악하기 좋았으나,
모두의 커밋을 합치고 보니 깃 히스토리가 길어져 작업 상황을 한 눈에 보기 어렵다는 단점이 있었습니다.

그래서 메인 브랜치는 squash로, 피쳐 브랜치는 지금처럼 merge 방식으로하면 어떨까 하는데, 어려분의 생각은 어떤가요?

**Comment by anyl92** [comment]
오.. 저는 승원님과 반대로 생각 했었는데요,
목표는 깔끔하게 작업 상황을 보자라는 것이 동일하지만

피쳐 브랜치에서 세세한 작업을 아토믹커밋으로 많이 올리다보니 피쳐 브랜치를 squash로,
메인 브랜치는 깔끔하게 유의미한 작업들을 볼 수 있게 commit merge 방식으로 하면 어떨까 생각했어요..

하지만... 자칫하다 실수하면 어떡하나의 두려움에 ㅋㅋㅋ 혼자만 생각했답니다..

**Reply by salmontaker** [reply]
저희에겐 git 마스터 @SeungrokYoon님이 계시니까 걱정은 하지 않아도 될 것 같아요 ㅋㅋ

다른 분들 의견도 들어보고 회의때 논의해 봅시다!

**Reply by SeungrokYoon** [reply]
저는 

피처 브랜치에서 메인으로 머지할 때는 `rebase merge`
그리고 서브 피처 브랜치에서 피처 브랜치로 머지할 때는 `squash merge`방식을 사용하면 좋을 것 같다는 생각을 간밤에 했습니다.

피처 브랜치에서 메인으로 머지할 때 리베이스 머지를 사용하고 싶은 이유는, 현재 저희가 리베이스 후 머지커밋 방식을 쓰고 있기에 그대로 사용하는 것이고, 피처 브랜치의 히스토리를 명료히 보이게 함입니다.

서브 피처 브랜치에서 피처 브랜치로 머지할 때 스쿼시 머지를 사용하고 싶은 이유는, sub feature 에는 너무 작은 커밋들이 있어서 이것들이 합쳐져도 사실 피처 브랜치의 작은 일부분이 될 것이기 때문입니다.

만약 서브 피처 브랜치에 많은 커밋이 들어있다면, 그것은 작업 범위를 너무 크게 설정한 것이라 생각해요. 

다른 분들은 어떻게 생각하시는지 궁금합니다 ㅎㅎ
